sycl/include/CL/sycl/handler.hpp

```diff
@@ -286,6 +286,22 @@ class handler {
       case access::target::local: {
         detail::LocalAccessorBaseHost *LAcc =
             static_cast<detail::LocalAccessorBaseHost *>(Ptr);
+        // Stream implementation creates local accessor with size per work item
+        // in work group. Number of work items is not available during stream
+        // construction, that is why size of the accessor is updated here using
+        // information about number of work items in the work group.
+        if (detail::getSyclObjImpl(*LAcc)->PerWI &&
+            MNDRDesc.GlobalSize.size() != 1 &&
+            MNDRDesc.GlobalSize.size() != 1) {
+          auto LocalAccImpl = detail::getSyclObjImpl(*LAcc);
+          // If local size is not speicified then consider that there is one
```

Comment:
So if user submits a kernel with 1M work-items, and let the RT decide the number of WI in a WG, we're going to allocate local memory as if there were 1M work-items in a workgroup?
That seems not very scalable.

Reply:
Changed this part. As you said, if user doesn't provide WG size then RT chooses number of WIs in WG. SO we will know this number only during kernel execution in __init. So if WG size is not provided by user, I use global size or info about max WG size for device to allocate local memory (take the minimum).  In the worst case decided to allocate 80% of local mem.